### PR TITLE
PIM-5269: fix date filter on product grid

### PR DESCRIPTION
**Pull request by juliensnz**
| Q                 | A
| ----------------- | ---
| Specs             | NA
| Behats            | NA
| Blue CI           | Y
| Changelog updated | Y
| Review and 2 GTM  | Y

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ function($, _, __, ChoiceFilter, localeSettings) {
          *
          * @property
          */
-        dateWidgetSelector: 'div#ui-datepicker-div.ui-datepicker',
+        dateWidgetSelector: 'div.datepicker',
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: div is useless

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 2]
```diff
@@ -248,8 +248,8 @@ function($, _, __, ChoiceFilter, localeSettings) {
          * @inheritDoc
          */
         _formatDisplayValue: function(value) {
-            var fromFormat = this.dateWidgetOptions.altFormat;
-            var toFormat = this.dateWidgetOptions.dateFormat;
+            var toFormat = this.dateWidgetOptions.format;
+            var fromFormat = this.dateWidgetOptions.defaultFormat;
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: Why switch the lines ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 3]
```diff
@@ -339,21 +334,7 @@ function($, _, __, ChoiceFilter, localeSettings) {
...
-            newValue = newValue.value;
-            oldValue = oldValue.value;
-
-            if ((newValue && (newValue.start || newValue.end)) ||
-                (oldValue && (oldValue.start || oldValue.end))
-            ) {
-                this.trigger('update');
-            }
-        },
+        _triggerUpdate: function(newValue, oldValue) {},
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: Not used anymore ? Why keep it ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 4]
```diff
@@ -382,6 +366,41 @@ function($, _, __, ChoiceFilter, localeSettings) {
...
+            this.trigger('update');
+
+            return this;
+        },
+
+        _onClickOutsideCriteria: function(e) {
+            var elem = this.$(this.criteriaSelector);
+            var isDatepicker = false;
+            $.each(e.originalEvent.path, function (key, value) {
+               if (true === $(value).hasClass('datepicker')) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: `true ===` is useless

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js [hunk 5]
```diff
@@ -22,12 +63,121 @@ define(['jquery', 'underscore', 'oro/datafilter/date-filter', 'oro/locale-settin
...
             altSeparator: ' ',
-            altTimeFormat: 'HH:mm'
-        }, DateFilter.prototype.dateWidgetOptions),
+            altTimeFormat: 'HH:mm',
+            changeMonth: true,
+            changeYear:  true,
+            yearRange:  '-80:+80',
+            dateFormat: localeSettings.getVendorDateTimeFormat('jquery_ui', 'date', 'mm/dd/yy'),
+            altFormat:  'yy-mm-dd',
+            className:      'date-filter-widget',
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js]: nice indentation

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js [hunk 6]
```diff
@@ -65,6 +267,128 @@ define(['jquery', 'underscore', 'oro/datafilter/date-filter', 'oro/locale-settin
         },
 
         /**
+         * Format datetes in a valut to another format
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js]: wtf ?

**Comment by phaseinducer** [comment]
GTM after fix (or not)

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js (hunk 1 above)
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: why ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js (hunk 3 above)
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: maybe in parent ?

Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: yep, if you remove it, each time you click on a date, datagrid is refreshed, datepicker is closed (by parent choice-filter)

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js [hunk 7]
```diff
@@ -1,15 +1,56 @@
 /* global define */
-define(['jquery', 'underscore', 'oro/datafilter/date-filter', 'oro/locale-settings'],
-    function($, _, DateFilter, localeSettings) {
+define(['jquery', 'underscore', 'oro/translator', 'oro/datafilter/choice-filter', 'oro/locale-settings'],
+function($, _, __, ChoiceFilter, localeSettings) {
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js]: as you removed `localeSettings` in date-filter, can you remove it too ?

**Comment by skeleton** [comment]
tiny comment, then GTM

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js (hunk 7 above)
Comment (ClementGautier) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/datetime-filter.js]: looks like it doesn't work anymore without it. It's actually used.

**Comment by ClementGautier** [comment]
Not GTM untill the behats are fixed. Actually it does not work

**Pull request merged**